### PR TITLE
File sending

### PR DESCRIPTION
**Pull request by himanish-k**
- Added functionality to transfer files between two peers.

### CODE REVIEW

**Review by michaelfakhri — COMMENTED** [review]
Do you think we need support for text messages? This PR removes support for text messages? We need a way to differentiate between text messages and files. Otherwise we cant support queries

**Comment by himanish-k** [comment]
It supports both text message and file sending and differentiates between the two as well. 

But I think we should wait until tomorrow before merging because I can still refine the code and the user interface a little more.

**Comment by michaelfakhri** [comment]
well I cant confirm or reject your comment since sending text messages is not even working, I just tried your latest code and when I send text, I get a syntax error on the receiving side. Could you have a look? Either way I only see one path for the code where the text messages and files are intermingled without any distinction. Here is where I think it will fail (if it were working in the first place), you would send a text message followed by a file. The text message would be combined with the file corrupting it since you oush all messages to a single array. I will make a suggestion in the code

File: peerConnection.js [hunk 1]
```diff
@@ -76,16 +77,67 @@ function configureChannel(channel) {
...
 	console.log("Sent message to PEER: " + msg);
 	this.channel.send(msg);
 }
 
+var arrayToStoreChunks = [];
 function messageReceiver(event) {
-	// For now this is only text but files should be implemented soon
-	console.debug("Got message from PEER: " + event.data,"");
+
+    var data = JSON.parse(event.data);
```
Comment (michaelfakhri) [inline peerConnection.js:124]: you could try catch statement here

File: peerConnection.js [hunk 2]
```diff
@@ -76,16 +77,67 @@ function configureChannel(channel) {
...
+	console.log("Got message from PEER: "+ data.message, event);
+
+	arrayToStoreChunks.push(data.message); // pushing chunks in array
```
Comment (michaelfakhri) [inline peerConnection.js]: check here if(data.message){
.... proceed to process file data
}

**Review by michaelfakhri — COMMENTED** [review]
Here are a few suggestions you use or not.

**Comment by himanish-k** [comment]
So I resolved the errors that you mentioned above. It's able to send files and instant messages now. Here's what I want to work on next, unless it seems unnecessary or should be deferred to the next version:
1.  Ability to save file with original file name.
2. Improve UI so browser prompts user to download file instant of creating a link. 
3. Improve UI so that instant messages show up as notifications, not just as messages in the browser console.
4. I want to define a universal Packet object that serves as the basic medium of exchange between peers. The way that I differentiate between an instant message and a file is by adding a 'type' property to the packet being sent; this property is 'file' for files and 'im' for instant messages. We may want to add other properties to the packet later as well when exchanging information gets more complex, so defining such an object seems appropriate.
5. There could be an error in a scenario where a very large file is being sent over a few seconds and an instant message is sent simultaneously. This IM will probably get added to the file.
6. The above issue bascially hints that currently simultaneous exchange of multiple files/IMs between peers is not supported yet.

**Comment by himanish-k** [comment]
Feedback from Michael:
1. Does not contribute to final goal which is building on content addressable P2P network.
2. Additional feature but good to have.
3. IM will bascially be queries. So, it doesn't make sense to display them.
4. Approved but efficiency should be maintained.
5. Not applicable.
6. Not applicable.

**Pull request merged**